lib/irb/type_completion/completor.rb

```diff
@@ -0,0 +1,230 @@
+require 'prism'
+require 'irb/completion'
+require_relative 'type_analyzer'
+module IRB
+  module TypeCompletion
+    class Completor < BaseCompletor # :nodoc:
+      HIDDEN_METHODS = %w[Namespace TypeName] # defined by rbs, should be hidden
+
+      def inspect
+        name = 'TypeCompletion::Completor'
+        prism_info = "Prism: #{Prism::VERSION}"
+        if Types.rbs_builder
+          "#{name}(#{prism_info}, RBS: #{RBS::VERSION})"
+        elsif Types.rbs_load_error
+          "#{name}(#{prism_info}, RBS: #{Types.rbs_load_error.inspect})"
+        else
+          "#{name}(#{prism_info}, RBS: loading)"
+        end
+      end
+
+      def completion_candidates(preposing, target, _postposing, bind:)
+        @preposing = preposing
+        verbose, $VERBOSE = $VERBOSE, nil
+        code = "#{preposing}#{target}"
+        @result = analyze code, bind
+        name, candidates = candidates_from_result(@result)
+
+        all_symbols_pattern = /\A[ -\/:-@\[-`\{-~]*\z/
+        candidates.map(&:to_s).select { !_1.match?(all_symbols_pattern) && _1.start_with?(name) }.uniq.sort.map do
+          target + _1[name.size..]
+        end
+      rescue SyntaxError, StandardError => e
+        handle_error(e)
+        []
+      ensure
+        $VERBOSE = verbose
+      end
+
+      def doc_namespace(preposing, matched, postposing, bind:)
+        name = matched[/[a-zA-Z_0-9]*[!?=]?\z/]
+        method_doc = -> type do
+          type = type.types.find { _1.all_methods.include? name.to_sym }
+          if type.is_a? Types::SingletonType
```

Comment:
At line 60 you used case statement for the exact conditions. Let's be consistent and do the same here too.